deno_dist/helper/ssg/index.ts

```diff
@@ -78,17 +78,41 @@ export const fetchRoutesContent = async <
   const baseURL = 'http://localhost'
 
   for (const route of inspectRoutes(app)) {
-    if (route.isMiddleware) continue
+    if (route.isMiddleware || isDynamicRoute(route.path)) continue
 
-    const url = new URL(route.path, baseURL).toString()
+    const contentData = await fetchRouteContent(app, route.path, baseURL)
+    if (contentData) {
+      htmlMap.set(route.path, contentData)
+    }
+  }
+
+  return htmlMap
+}
+
+interface RouteContent {
+  content: string | ArrayBuffer
+  mimeType: string
+}
+
+const fetchRouteContent = async <E extends Env = Env, S extends Schema = {}>(
+  app: Hono<E, S>,
+  path: string,
+  baseURL: string
+): Promise<RouteContent | null> => {
+  try {
+    const url = new URL(path, baseURL).toString()
     const response = await app.fetch(new Request(url))
     const mimeType = response.headers.get('Content-Type')?.split(';')[0] || 'text/plain'
     const content = await parseResponseContent(response)
-
-    htmlMap.set(route.path, { content, mimeType })
+    return { content, mimeType }
+  } catch (error) {
+    console.error(`Error fetching content for path: ${path}`, error)
+    return null
   }
+}
 
-  return htmlMap
+const isDynamicRoute = (path: string): boolean => {
+  return /:\w+|\*/.test(path)
```

Comment:
The determination of the named path parameters here is incorrect.

`/foo:bar` is not a dynamic route. You can see the spec here:

https://github.com/honojs/hono/blob/a00fc566e93374acd4c8999443f4b48f79da05b9/src/utils/url.ts#L55

So you have to split the path with `/` and determine if there is a `:` at the beginning of it or not.

Reply:
fixed.